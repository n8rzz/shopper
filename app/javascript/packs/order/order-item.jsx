import React from 'react';
import PropTypes from 'prop-types'
import classnames from 'classnames';
import {noop} from 'lodash/noop';

export class OrderItem extends React.Component {
    constructor(props) {
        super(props);
    }

    _buildOrderItemClassnames() {
        const orderItemIdClassname = `js-orderItem-${this.props.orderItemId}`;

        return classnames({
            'orderItem': true,
            [orderItemIdClassname]: true,
            'mix-orderItem_isPicked': this.props.isPicked,
        });
    }

    _buildCheckboxJsx() {
        return (
            <div className="orderItem-action">
                <form>
                    <input
                        type="checkbox"
                        className="input-checkbox js-orderItem-action-value u-isVisuallyHidden"
                        name={this.props.orderItemId}
                        checked={this.props.isPicked}
                        onChange={() => noop}
                    />
                    <label
                        className="input-label_checkbox js-orderItem-action"
                        htmlFor={this.props.orderItemId}
                        onClick={this.props.onClickCheckboxHandler}
                    >
                        {this.props.isPicked}
                    </label>
                </form>
            </div>
        );
    }

    _buildItemTitleJsx() {
        return (
            <div className="orderItem-hd-content">
                <h3>
                    <a
                        href={this.props.editItemUrl}
                        className="orderItem-hd-link"
                    >
                        {this.props.itemName}
                    </a>
                </h3>
            </div>
        );
    }

    _buildQtyJsx() {
        if (this.props.qty < 2) {
            return null;
        }

        return (
            <div className="orderItem-hd-supplement">
                <h3>
                    <span className="txt-light">Qty:</span> {this.props.qty}
                </h3>
            </div>
        );
    }

    _buildRemoveItemButtonJsx() {
        return (
            <button
                type="submit"
                className="icon-btn js-orderItem-remove"
                onClick={this.props.onClickRemoveItemHandler}
            >
                <span className="icon icon-trashcan"></span>
            </button>
        );
    }

    _buildAssemblyTitleJsx() {
        if (this.props.assemblyName === '') {
            return null;
        }

        return (
            <div className="orderItem-bd">
                {this.props.assemblyName}
            </div>
        );
    }

    render() {
        return (
            <li className={this._buildOrderItemClassnames()}>
                {this._buildCheckboxJsx()}

                <div className="orderItem-content">
                    <div className="orderItem-hd">
                        {this._buildItemTitleJsx()}
                        {this._buildQtyJsx()}
                        {this._buildRemoveItemButtonJsx()}
                    </div>
                    {this._buildAssemblyTitleJsx()}
                    <div className="orderItem-ft">
                        {this.props.departmentName}
                    </div>
                </div>
            </li>
        );
    }
}

OrderItem.propTypes = {
    assemblyName: PropTypes.string.isRequired,
    departmentName: PropTypes.string.isRequired,
    editItemUrl: PropTypes.string.isRequired,
    isPicked: PropTypes.bool.isRequired,
    itemName: PropTypes.string.isRequired,
    orderItemId: PropTypes.number.isRequired,
    qty: PropTypes.number.isRequired,
    onClickCheckboxHandler: PropTypes.func.isRequired,
    onClickRemoveItemHandler: PropTypes.func.isRequired,
};
