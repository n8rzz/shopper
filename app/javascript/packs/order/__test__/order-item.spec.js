import React from 'react';
import {shallow} from 'enzyme';

import { OrderItem } from '../order-item';

describe('OrderItem', () => {
    let onClickCheckboxHandlerSpy;
    let onClickRemoveItemHandlerSpy;

    beforeEach(() => {
        onClickCheckboxHandlerSpy = jest.fn();
        onClickRemoveItemHandlerSpy = jest.fn();
    });

    afterEach(() => {
        onClickCheckboxHandlerSpy = null;
        onClickRemoveItemHandlerSpy = null;
    });

    describe('when passed an empty `#assemblyName`', () => {
        test('renders correctly with valid props without ', () => {
            const component = shallow(<OrderItem
                assemblyName={''}
                csrf={'3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=='}
                departmentId={32}
                departmentName={'meat'}
                editItemUrl={'http://localhost:3000/orders/21/edit'}
                isPicked={false}
                itemName={'$texas'}
                itemName={'Red Face Tuna'}
                orderItemId={11}
                qty={1}
                onClickCheckboxHandler={onClickCheckboxHandlerSpy}
                onClickRemoveItemHandler={onClickRemoveItemHandlerSpy}
            />);

            expect(component).toMatchSnapshot();
        });
    });

    describe('when passed a non-empty `#assemblyName`', () => {
        test('renders correctly with valid props without ', () => {
            const component = shallow(<OrderItem
                assemblyName={'sushi'}
                csrf={'3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=='}
                departmentId={32}
                departmentName={'meat'}
                editItemUrl={'http://localhost:3000/orders/21/edit'}
                isPicked={false}
                itemName={'$texas'}
                itemName={'Red Face Tuna'}
                orderItemId={11}
                qty={1}
                onClickCheckboxHandler={onClickCheckboxHandlerSpy}
                onClickRemoveItemHandler={onClickRemoveItemHandlerSpy}
            />);

            expect(component).toMatchSnapshot();
        });
    });
});
