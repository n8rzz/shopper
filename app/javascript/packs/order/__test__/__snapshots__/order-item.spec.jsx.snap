// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem when passed a non-empty \`#assemblyName\` renders correctly with valid props 1`] = `
<li
  className="orderItem js-orderItem-11"
>
  <div
    className="orderItem-action"
  >
    <form>
      <input
        checked={false}
        className="input-checkbox js-orderItem-action-value u-isVisuallyHidden"
        name={11}
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="input-label_checkbox js-orderItem-action"
        htmlFor={11}
        onClick={[MockFunction]}
      />
    </form>
  </div>
  <div
    className="orderItem-content"
  >
    <div
      className="orderItem-hd"
    >
      <div
        className="orderItem-hd-content"
      >
        <h3>
          <a
            className="orderItem-hd-link"
            href="http://localhost:3000/orders/21/edit"
          >
            Red Face Tuna
          </a>
        </h3>
      </div>
      <button
        className="icon-btn js-orderItem-remove"
        onClick={[MockFunction]}
        type="submit"
      >
        <span
          className="icon icon-trashcan"
        />
      </button>
    </div>
    <div
      className="orderItem-bd"
    >
      sushi
    </div>
    <div
      className="orderItem-ft"
    >
      meat
    </div>
  </div>
</li>
`;

exports[`OrderItem when passed an empty \`#assemblyName\` renders correctly with valid props 1`] = `
<li
  className="orderItem js-orderItem-11"
>
  <div
    className="orderItem-action"
  >
    <form>
      <input
        checked={false}
        className="input-checkbox js-orderItem-action-value u-isVisuallyHidden"
        name={11}
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="input-label_checkbox js-orderItem-action"
        htmlFor={11}
        onClick={[MockFunction]}
      />
    </form>
  </div>
  <div
    className="orderItem-content"
  >
    <div
      className="orderItem-hd"
    >
      <div
        className="orderItem-hd-content"
      >
        <h3>
          <a
            className="orderItem-hd-link"
            href="http://localhost:3000/orders/21/edit"
          >
            Red Face Tuna
          </a>
        </h3>
      </div>
      <button
        className="icon-btn js-orderItem-remove"
        onClick={[MockFunction]}
        type="submit"
      >
        <span
          className="icon icon-trashcan"
        />
      </button>
    </div>
    <div
      className="orderItem-ft"
    >
      meat
    </div>
  </div>
</li>
`;
